lib/ae_mdw/node/db.ex

```diff
@@ -16,21 +16,42 @@ defmodule AeMdw.Node.Db do
   @typep height_hash() :: {hash_type(), pos_integer(), binary()}
 
   @spec get_blocks(Blocks.block_hash(), Blocks.block_hash()) :: tuple()
-  def get_blocks(kb_hash, next_gen_kb_hash) do
-    {:aec_db.get_block(kb_hash), get_micro_blocks(next_gen_kb_hash)}
+  def get_blocks(kb_hash, next_kb_hash) do
+    {:aec_db.get_block(kb_hash), get_micro_blocks(next_kb_hash)}
   end
 
   @spec get_micro_blocks(Blocks.block_hash()) :: list()
-  def get_micro_blocks(next_gen_kb_hash) do
-    next_gen_kb_hash
+  def get_micro_blocks(next_kb_hash) do
+    next_kb_hash
     |> :aec_db.get_header()
     |> :aec_headers.prev_hash()
     |> Stream.unfold(&micro_block_walker/1)
     |> Enum.reverse()
   end
 
-  @spec get_tx_data(binary()) :: tuple()
+  @spec get_next_hash(Blocks.block_hash(), Blocks.mbi()) :: Blocks.block_hash()
+  def get_next_hash(next_kb_hash, mbi) do
+    next_kb_hash
+    |> get_micro_blocks()
+    |> Enum.reduce_while(0, fn mblock, index ->
+      if index == mbi + 1 do
+        ok_mb_hash =
+          mblock
+          |> :aec_blocks.to_micro_header()
+          |> :aec_headers.hash_header()
+
+        {:halt, ok_mb_hash}
+      else
+        {:cont, index + 1}
+      end
+    end)
+    |> case do
+      {:ok, mb_hash} -> mb_hash
+      _mb_count -> next_kb_hash
+    end
+  end
 
+  @spec get_tx_data(binary()) :: tuple()
```

Comment:
I think we can turn this into
```elixir
(Txs.tx_hash()) :: {Blocks.block_hash(), Node.tx_type(), Node.aext(), Node.tx()}
```

Reply:
This is preexisting, it will be improved in a following PR. Thanks